presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -359,6 +360,8 @@ public ActualProperties visitJoin(JoinNode node, List<ActualProperties> inputPro
             ActualProperties probeProperties = inputProperties.get(0);
             ActualProperties buildProperties = inputProperties.get(1);
 
+            boolean unordered = spillPossible(session, node.getType());
```

Comment:
Why doesn't spill preserve any grouping or sorting local properties?

Reply:
Spill works like "hybrid hash join" - some build side's partitions are in memory and some can be spilled to disk. Probe rows that match in-memory partitions are processed normally and those matching on-disk partitions are spilled, to be processed later. This changes order, so breaks sorting.

When it comes to grouping, from reading the code I understood "grouping over _s_" means "rows having same value of _s_ are next to each other". So reordering of rows breaks grouping. Note that even when _s_ was a join symbol, reodering can happen, since spill state is dynamic. Even single probe row that produces 100 output rows can produce 50 rows first and 50 rows later.